packages/server/integrations/updatePrevUsedRepoIntegrationsCache.ts

```diff
@@ -0,0 +1,31 @@
+import IntegrationRepoId from 'parabol-client/shared/gqlIds/IntegrationRepoId'
+import getAllCachedRepoIntegrations from '../graphql/queries/helpers/getAllCachedRepoIntegrations'
+import getPrevUsedRepoIntegrations from '../graphql/queries/helpers/getPrevUsedRepoIntegrations'
+import getPrevUsedRepoIntegrationsRedisKey from '../utils/getPrevUsedRepoIntegrationsRedisKey'
+import getRedis from '../utils/getRedis'
+
+const updatePrevUsedRepoIntegrationsCache = async (teamId: string, repoIntegrationId: string) => {
```

Comment:
> is it possible that other teammates would see a list of my repositories instead of their own? 

Yeah, I think that's a valid tension, but that's how the integrations are currently working in prod. If one user adds the GitHub integration, we can all see their repos in a Sprint Poker meeting. That might feel strange to some users, but it does mean that each team member doesn't need to add an integration to use the meeting - just one person does 

> will teammates clear each other caches?

Yep, they will as the cache is by team. I think that's the right move as everyone on the team should have access to the team's repo that they want to use

Reply:
yaaa i think the thinking was (back in the day) that if a user didn't have access to write to a certain repo, they could still push to it & we'd use the other person's credentials.
for example, if Nick had access to a repo I didn't, I could push it it & then it would show Nick adding a task & there'd be a footer saying "on behalf of Matt". ...tbh I'm not sure if we kept that logic or not :sweat_smile:

Reply:
My concern is that 2 users, even if they both have an access to the same repo, will have different set of repos. So for the second user I would see cached **private repos** of the first user. He might not want to expose it.

What do you think?

Reply:
I tried to test GitHub integrations with 2 different users with 2 different GitHub  accounts. I notice few issues and tried to explain the privacy issue. Here is the loom — https://www.loom.com/share/5cb94e6f8a97495aa568d0e2be5c9e74 

In case of poker I unable to see your private repos, however in case of shared cache I will able to

<img width="1575" alt="image" src="https://user-images.githubusercontent.com/466991/195926407-68f0996d-9498-4de4-9b9c-e248a695ce5e.png">

What do you think? Are you able to reproduce issues from the loom?

Reply:
> For the bug you mentioned where the user is unable to see any issues in Poker, does your second GitHub account have any issues?

@nickoferrall second account only have an access to Parabol repo, but it have no issues created by himself on GitHub. Is that what you mean?

Reply:
Does the second GitHub account belong to any repos that contain issues? Does it belong to the Parabol repo, for instance?

I tested the issue by creating a brand new GitHub account and I created a couple of repos. I integrated with Parabol, but couldn't see any issues in Sprint Poker (just like the bug you mentioned), but that was because I hadn't added any issues to that repo.

If your second GitHub account does belong to a repo that contains issues, would you mind double-checking if the bug persists after my latest changes?